### PR TITLE
BaSP-TG-34: List/Delete Timesheet Page

### PR DESCRIPTION
**Pull request by Eju97**
- It contains 2 components:
         1. Main component that contains the list in table form.
         2. Delete component inside a modal that executes the DELETE method.

Even though the functionalities of this PR are ready, some styles are yet to be done.

### CODE REVIEW

File: src/Components/TimeSheets/index.js [hunk 1]
```diff
@@ -1,9 +1,86 @@
+import { useState, useEffect } from 'react';
 import styles from './time-sheets.module.css';
+import Modal from './ModalDelete/modalDelete';
 
 function TimeSheets() {
```
Comment (lmalvasia) [inline src/Components/TimeSheets/index.js:5]:
```suggestion
const TimeSheets = () => {
```

File: src/Components/TimeSheets/ModalDelete/modalDelete.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import styles from './modalDelete.module.css';
+
+const TimesheetDeleter = (props) => {
```
Comment (lmalvasia) [inline src/Components/TimeSheets/ModalDelete/modalDelete.js:3]:
```suggestion
const DeleteConfirmationModal = (props) => {
```

File: src/Components/TimeSheets/index.js [hunk 3]
```diff
@@ -1,9 +1,86 @@
+import { useState, useEffect } from 'react';
 import styles from './time-sheets.module.css';
+import Modal from './ModalDelete/modalDelete';
```
Comment (lmalvasia) [inline src/Components/TimeSheets/index.js:3]:
```suggestion
import DeleteConfirmationModal from './DeleteConfirmationModal/DeleteConfirmationModal';
```

**Comment by eyrent** [comment]
### QA Report
#### Functionality tested
- Timesheets are listed in a table as expected ✅
- Timesheets are deleted as expected (from the DB, as well) ✅
#### What can be improved?
:yellow_circle: When loading timesheets and there's no data to populate the table, one can see a _There are no Timesheets_ **message that can be changed to a 'loading' message**, reserving the former for an empty list only. _[might be assigned as a new ticket]_
:yellow_circle: **Dates are shown in ISO 8601 format** straight from MongoDB. Although this was not included in the ticket it'll be nice if this could be **changed to a friendlier format**. Also **time is not necessary** (hours, minutes, seconds and timezone) and are better omitted, since the column name is 'Date'. _[might be assigned as a new ticket]_
:large_blue_circle: Even though styles were not part of this PR, I want to remind you to **use column lines separators** so that cell text can't be misunderstood. 
:large_blue_circle: The deletion confirmation **modal won't go away when clicking outside** (not in any spec, but it's an usual behavior). _[might be assigned as a new ticket]_
:green_circle:  I checked to see if employee `NO ME BORRES` was **listed among the rest of the employees but that was not** the case. This looks like a mistake setting up the testing environment (database documents, more precisely), and will be tested again when other components are available. _[Check this]_

I'll be consulting if new tickets need to be made or if the recommendations I left are of any priority at the moment (probable not).  But I'm giving you a GO for functionality asked in the ticket associated with this PR. Good luck merging!

**Comment by Tibor700** [comment]
Everything looks fine , the functionalities of the cheks of the trello cards works as expected, the only change i recomend is the date format of start date and end date ,to be easier to read and understand like month/day/year.

**Pull request merged**